h3/src/server.rs

```diff
@@ -467,6 +515,102 @@ where
     }
 }
 
+/// Configures the HTTP/3 connection
+#[derive(Debug, Clone, Copy)]
+pub struct Config {
```

Comment:
At the moment there are 2 ways to create a `Connection`. One to use a this `Config` struct and a second using `Builder` (which uses this struct under the hood). I think there should be only one way to create a new Connection. 

The current `Builder`s have the advantage to be able to support different configurations for client and server if needed.
For the common stuff between client and server is is fine to use this struct. But then is should probably get its own file.

Reply:
I see so the ask here is to continue using the builder pattern as opposed to passing a config object. 

This means that all properties in the config object have to be moved to `Connection`

Reply:
Not quite, the `Connection` can still have the `config` object. There was 2 ways to create a connection.
1. Create a `Config` with all necessary properties and construct a `Connection` with this `Config`
2. Use a builder which constructs the `Config` and the `Connection`

@ten3roberts already removed option one in [here](https://github.com/hyperium/h3/pull/183/commits/3cb795546f50961c3bd7030d7015333e4a55525b)
Yet there is only one way to create a `Connection` using the Builder. 
Since the `Client` also uses the `Config` struct, the only thing left is to move `Config` out of the `server.rs` file. IMO it would be best to create a `config.rs` file.

Reply:
If I understand it all corretly you think it is fine to keep the `Config` struct but only use it internally, if that is correct?

I moved the fields inside the builder into a separate `Config` struct, because with all the options for webtransport, connect, datagrams, etc it became cumbersome to pass all those around separately and then have as "inlined" fields in the resulting connection.